fiber/src/fiber.mli

```diff
@@ -106,10 +106,20 @@ val parallel_iter_seq : 'a Seq.t -> f:('a -> unit t) -> unit t
 val sequential_iter_seq : 'a Seq.t -> f:('a -> unit t) -> unit t
 
 (** Provide efficient parallel iter/map functions for maps. *)
-module Make_map_traversals (Map : Map.S) : sig
```

Comment:
Wondering about the old name: did we never provide `parallel_iter`?

Reply:
(The stake comment above should be updated.)

Reply:
It was just a wrapper around `Map.to_seq`, so there was no advantage to using it.

Reply:
Not sure I follow. How do you propose to get this functionality with the new API, via `parallel_map`?

Reply:
There's no need to get this functionality from the new API because it was already available without it :)
The old implementation was:

```ocaml
let parallel_iter t ~f =
    parallel_iter_seq (Map.to_seq t) ~f:(fun (k, v) -> f k v)
```

So all we were doing is wrapping `Map.to_seq`. IMO it's a bad API to require a user to provide all of `Map.S` where all we need is one function. And given that we don't even do anything interesting with this `to_seq`, it's basically just a pointless wrapper that can be eliminated to save the user from dealing with functors.

Reply:
Ah, of course, Fiber's `parallel_iter_seq` is sufficient to replicate the old behaviour. I suppose I was hoping for a more direct version that doesn't involve an intermediate `Seq.t` but there wasn't one before, so nothing really changed.

Reply:
> I suppose I was hoping for a more direct version that doesn't involve an intermediate Seq.t but there wasn't one before, so nothing really changed.

I was looking for one too. But unfortunately it doesn't seem possible with the map api from stdlib. Perhaps if we had access to the underlying binary representation we could do something better.